src/referral_maintenance.rs

```diff
@@ -0,0 +1,199 @@
+use crate::app_data_loading::load_distinct_app_data_from_json;
+use crate::models::app_data_json_format::AppData;
+use crate::models::referral_store::{Referral, ReferralStore};
+use anyhow::Result;
+use cid::Cid;
+use primitive_types::{H160, H256};
+use std::convert::TryFrom;
+use std::sync::Arc;
+use std::time::Duration;
+
+const MAINTENANCE_INTERVAL: Duration = Duration::from_secs(30);
+
+pub async fn referral_maintainance(memory_database: Arc<ReferralStore>, dune_data_folder: String) {
+    loop {
+        match maintenaince_tasks(Arc::clone(&memory_database), dune_data_folder.clone()).await {
+            Ok(_) => {}
+            Err(err) => tracing::debug!("Error during maintenaince_task for referral: {:?}", err),
+        }
+        tokio::time::sleep(MAINTENANCE_INTERVAL).await;
+    }
+}
+
+pub async fn maintenaince_tasks(db: Arc<ReferralStore>, dune_data_folder: String) -> Result<()> {
+    // 1st step: getting all possible app_data from file and store them in ReferralStore,
+    // if not yet existing
+    let vec_with_all_app_data = match load_distinct_app_data_from_json(
+        dune_data_folder + "app_data/distinct_app_data.json",
+    ) {
+        Ok(vec) => vec,
+        Err(err) => {
+            tracing::error!("Could not load distinct app data, due to: {:?}", err);
+            return Ok(());
+        }
+    };
+    for app_data in vec_with_all_app_data {
+        {
+            let mut guard = match db.0.lock() {
+                Ok(guard) => guard,
+                Err(poisoned) => poisoned.into_inner(),
```

Comment:
I think getting reference to the inner data if the mutex is poisoned is quite dangerous (as the HashMap may be in an undefined state). I would return an error in this case.